theatre/studio/src/panels/DetailPanel/propEditors/RgbaPropEditor.tsx

```diff
@@ -0,0 +1,124 @@
+import type {PropTypeConfig_Rgba} from '@theatre/core/propTypes'
+import type {Rgba} from '@theatre/shared/utils/color'
+import {
+  decorateRgba,
+  rgba2hex,
+  parseRgbaFromHex,
+} from '@theatre/shared/utils/color'
+import type SheetObject from '@theatre/core/sheetObjects/SheetObject'
+import React, {useCallback, useMemo, useRef} from 'react'
+import {useEditingToolsForPrimitiveProp} from './utils/useEditingToolsForPrimitiveProp'
+import {SingleRowPropEditor} from './utils/SingleRowPropEditor'
+import {debounce} from 'lodash-es'
+import {RgbaColorPicker} from 'react-colorful'
+import styled from 'styled-components'
+import usePopover from '@theatre/studio/uiComponents/Popover/usePopover'
+import BasicStringInput from '@theatre/studio/uiComponents/form/BasicStringInput'
+
+const RowContainer = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  gap: 4px;
+`
+
+const Puck = styled.div<{background: Rgba}>`
+  height: calc(100% - 4px);
+  aspect-ratio: 1;
+  background: ${(props) => rgba2hex(props.background)};
+  border-radius: 2px;
+`
+
+const HexInput = styled(BasicStringInput)`
+  flex: 1;
+`
+
+const noop = () => {}
+
+const RgbaPropEditor: React.FC<{
+  propConfig: PropTypeConfig_Rgba
+  pointerToProp: SheetObject['propsP']
+  obj: SheetObject
+}> = ({propConfig, pointerToProp, obj}) => {
+  const containerRef = useRef<HTMLDivElement>(null!)
+
+  const stuff = useEditingToolsForPrimitiveProp<Rgba>(
+    pointerToProp,
+    obj,
+    propConfig,
+  )
+
+  const debouncedPermanentlySetValue = useMemo(
+    () =>
+      debounce((rgba: Rgba) => {
```

Comment:
Probably an anti-pattern here. `temporarilySetValue` is there so we don't need to debounce and we can call it as often as we want. I think with debounce, there might be race conditions when sequence's position changes while the debouncer is waiting, which I think would create keyframe at the sequence's new position rather than at its original position

Reply:
I'm not debouncing calls to `temporarilySetValue`, but to `permanentlySetValue`, similarly to how debouncedScrub does it, since the current color picker doesn't let us easily see when the picking started/ended. I agree that the ad-hoc nature of this is a little lazy, the less lazy alternative to this would be putting a debounced scrub method on `stuff`. Not saying there couldn't be bugs here, but those bugs wouldn't probably also be replicated in debouncedScrub.

Reply:
Ah I see. Bad news is, `debouncedScrub()` also has that problem, and it's pretty annoying. We haven't shipped anything that uses `debouncedScrub()` (other than r3f, which isn't released yet) so I don't think users are encountering the problem.

I think it's a  good time to fix debouncedScrub and use it here, _or_ skip the debounce here. Otherwise => bad UX.

Reply:
Whatever bad ux we end up here is dwarfed by the bad UX of having an undo step at every color drag increment, so this seems like a fix, rather than a skip. Can you make me a minimal reproduction?

Reply:
So, for now, fork react-colorful and add temporary/permanentlySetValue callbacks to it. We do need to fix debounced scrub soon though.